callisto/delivery/models.py

```diff
@@ -257,8 +305,34 @@ def get_match(self, identifier):
             str or None: returns the decrypted report as a string if the identifier matches, or None otherwise.
         """
         decrypted_report = None
+        iterations = None
```

Comment:
all of this code is identical to the Report case, let's abstract it into a shared method or mixin